### PR TITLE
App cleanup

### PR DESCRIPTION
**Pull request by bluepilledgreat**
I don't trust myself with some of these changes. Please test thoroughly!

### CODE REVIEW

File: Bloxstrap/InterProcessLock.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bloxstrap
+{
```
[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/InterProcessLock.cs:8]: maybe, we dont need async mutexes right now

File: Bloxstrap/LaunchSettings.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+                    i++; // path arg
+                    RobloxLaunchArgs = $"-task EditFile -localPlaceFile \"{pathArg}\"";
+                }
+            }
+        }
+
+        private void Parse()
+        {
+            const string LOG_IDENT = "LaunchSettings::Parse";
```
[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/LaunchSettings.cs:133]: yup, i dont trust myself

File: Bloxstrap/App.xaml.cs [hunk 3]
```diff
@@ -163,19 +163,22 @@ protected override void OnStartup(StartupEventArgs e)
...
-                Process? menuProcess = Process.GetProcesses().Where(x => x.MainWindowTitle == $"{ProjectName} Menu").FirstOrDefault();
+                using InterProcessLock menuLock = new InterProcessLock("Menu", TimeSpan.FromMilliseconds(100));
 
-                if (menuProcess is not null)
+                if (!menuLock.IsAcquired)
                 {
-                    var handle = menuProcess.MainWindowHandle;
-                    Logger.WriteLine(LOG_IDENT, $"Found an already existing menu window with handle {handle}");
-                    PInvoke.SetForegroundWindow((HWND)handle);
+                    Frontend.ShowMessageBox(
```
[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: i feel like people would be confused to why the menu isnt opening.
added back the log and added in a quiet flag check.

[the remark this answers is not in the document]
Comment (pizzaboxer) [reply Bloxstrap/App.xaml.cs]: if you already have a menu instance open then it should be fairly obvious shouldn't it lol

[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: ive opened the bloxstrap menu without noticing i already had another instance open on multiple occasions lol

[the remark this answers is not in the document]
Comment (pizzaboxer) [reply Bloxstrap/App.xaml.cs]: in which case bring the already existing menu to the foreground

[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: if we cant find the existing menu, do we show the error message?

[the remark this answers is not in the document]
Comment (pizzaboxer) [reply Bloxstrap/App.xaml.cs]: that's a good point, there doesn't appear to be a way to identify what process is holding a mutex in c#?

[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: cant find a way in c#, we'd have to do more research

[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: lots of win32 api shit, i aint doing that. @pizzaboxer if you're down for a small qol feature, go for it.

[the remark this answers is not in the document]
Comment (pizzaboxer) [reply Bloxstrap/App.xaml.cs]: are you sure you dont just want to stick to detecting window titlebars lol

**Comment by d44df016fe28bc8d5974** [comment]
new version just dropped

File: Bloxstrap/InterProcessLock.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pizzaboxer) [reply Bloxstrap/InterProcessLock.cs:8]: we might need to if we still need to use it in the bootstrapper thread

File: Bloxstrap/App.xaml.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: using mutexes is way cleaner imo

[the remark this answers is not in the document]
Comment (pizzaboxer) [reply Bloxstrap/App.xaml.cs]: sure i guess but i honestly don't see much problem with the approach we currently take, and it does what we want it to do

[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: do you want me to completely revert mutex stuff or do you want me to use the old approach of finding the menu and setting it as the foreground

[the remark this answers is not in the document]
Comment (pizzaboxer) [reply Bloxstrap/App.xaml.cs]: keep the mutex helper utility, that could be useful, but i do think the old approach for handling duplicate menu instances was better

[the remark this answers is not in the document]
Comment (bluepilledgreat) [reply Bloxstrap/App.xaml.cs]: aight will do

**Review by pizzaboxer — APPROVED** [review]
lgtm

**Comment by bluepilledgreat** [comment]
> lgtm

who needs testing anyways :)

**Pull request merged**